tests/unittest/noctx.cpp

```diff
@@ -0,0 +1,149 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2025 Intel Corporation
+ */
+
+#include "noctx.hpp"
+
+void NoCtxTest::SetUp() {
+  ctx = new struct st_tests_context;
+
+  /* NOCTX test: always operate on a copy of the global ctx.
+     Do not use the global ctx directly for anything except copying its values.
+   */
+  ASSERT_TRUE(ctx != nullptr);
+  memcpy(ctx, st_test_ctx(), sizeof(*ctx));
+
+  ASSERT_TRUE(ctx->handle == nullptr)
+      << "NOCTX tests should be run wiht --no_ctx_tests flag!";
+
+  ctx->level = ST_TEST_LEVEL_MANDATORY;
+  ctx->para.flags |= MTL_FLAG_RANDOM_SRC_PORT;
+  ctx->para.log_level = MTL_LOG_LEVEL_INFO;
+  ctx->para.priv = ctx;
+  ctx->para.tx_queues_cnt[MTL_PORT_P] = 16;
+  ctx->para.tx_queues_cnt[MTL_PORT_R] = 16;
+  ctx->para.rx_queues_cnt[MTL_PORT_P] = 16;
+  ctx->para.rx_queues_cnt[MTL_PORT_R] = 16;
+  defaultTestDuration = 20;
+}
+
+void NoCtxTest::TearDown() {
+  mtl_stop(ctx->handle);
+  for (auto handler : st30pHandlers) {
+    if (handler) {
+      delete handler;
+      handler = nullptr;
+    }
+  }
+  st30pHandlers.clear();
```

Comment:
You are welcome to ignore that, but:
You could save objects directly in the vector or use smart pointer, then clear() will automatically clear everything.